parsers/MN.py

```diff
@@ -64,24 +124,13 @@ def fetch_consumption(
 ):
     if target_datetime:
         raise NotImplementedError("This parser is not yet able to parse past dates.")
-    with session.get(NDC_GENERATION) as response:
-        NDC_SOUP = BeautifulSoup(response.content, "html.parser")
-
-    # Нийлбэр ачаалал
-    consumption_MW = float(NDC_SOUP.find_all("td")[6].text.strip(" МВт"))
-
-    date_raw = NDC_SOUP.find_all("td")[0].text.strip("Хоногийн ачаалал: ")
-    date_pretty = datetime.strptime(date_raw, "%Y-%m-%d")
-    time_raw = NDC_SOUP.find_all("td")[14].text.strip("Импортын чадал ")
-    time_time = datetime.strptime(time_raw, "%H:%M:%S")
-    time_pretty = datetime.time(time_time)
 
-    time_combined = datetime.combine(date_pretty, time_pretty)
+    query_data = query(session)
 
     data = {
         "zoneKey": zone_key,
-        "datetime": time_combined.replace(tzinfo=timezone(TZ)),
-        "consumption": consumption_MW,
+        "datetime": query_data["time"],
+        "consumption": query_data["consumptionMW"],
```

Comment:
I don't know if you are aware of this but we have added parser classes with built in validation, it would be great if those could be implemented.

You can find the documentation here: https://github.com/electricitymaps/electricitymaps-contrib/wiki/Building-a-new-parser

If you don't feel comfortable doing it I'll do it when this PR is otherwise ready to be merged.

Reply:
d14dfca
I wasn't aware of that, but great to see a change to better Parser return objects!

Currently, even though this parser only returns one breakdown, I think the interface requires the `ProductionBreakdown` to be packed in a `ProductionBreakdownList`? Because the documentation says "In cases where the parser function collects data for more than one datetime, the data should be stored in the following [list] objects...", but returning only a single `ProductionBreakdown` gives an error.

Also since there is a ProductionBreakdown and a TotalProduction, I opted to use the breakdown, even though we only have solar/wind/unknown here. 

Feel free to change things if the new objects should be used differently :)

Reply:
Thanks for updating it!

So far things look good from a quick glance but I'll do a full review tomorrow morning.

As for the ProductionBreakdownList, this is required even if there is just a single data point so we can simplify things in the future.

As you noticed there is a to_list() function, in the future the goal is to remove the need for this and return the production breakdown list as is to enable us to further enhance validation and unify the parsers.

This also lays the groundwork so we can return multiple data types (consumption and production for example) in the same parser run making everything a bit more efficient (though this is not yet fully shaped).